sklearn/feature_extraction/text.py

```diff
@@ -473,6 +473,10 @@ def transform(self, X, y=None):
             Document-term matrix.
 
         """
+        if isinstance(X, str):
+            raise ValueError(
+                "iterable over raw text documents expected, string object received")
```

Comment:
Capitalize `iterable` and add a period? I'm not sure whether it'd be a good idea to include the given string in the error message, but maybe something to consider